module/Makefile.in

```diff
@@ -7,7 +7,7 @@ subdir-m += unicode
 subdir-m += zcommon
 subdir-m += zfs
 
-INSTALL_MOD_DIR ?= extra
+INSTALL_MOD_DIR ?= updates
```

Comment:
Why are we doing this instead of just adding a new directory and installing a config file in `/etc/depmod.d` for Ubuntu?

Believe it or not, there's actually a reason for installing into `extra`: RHEL and SLE weak module versioning kicks in for modules built targeting the `extra` path.

Reply:
@Conan-Kudo Can you elaborate a bit on the weak module versioning? I've tried reading online a bit, and from what I understand it means that it will look into the extra directories of previous kernel versions for modules that it doesn't find in the current kernel version, is that correct? If placing the modules into `updates` would break this behaviour on SLE and RHEL this would definitely be a blocker.

Reply:
Yes, you've hit the nail on the head.

Reply:
@behlendorf I think we should drop this PR in this case.

Reply:
Thanks for the review, it sounds like relocating from `extra` to `updates` will be a little more disruptive than I'd originally hoped.  After talking with @Conan-Kudo it sounds like the right way to handle this will be to add a new `--with-kmoddir=<extra|zfs>` configure option.

When set to **extra**, which should be the default, we want the behavior to remain unchanged.  This is the correct install location for RHEL, SLE, Debian, and others.  And it works correctly with weak-modules.

Ubuntu diverges from the established convention here by not including `extra` in the default search path.  And apparently we also shouldn't be using 'updates' either as it's intended for local one offs.  So @Conan-Kudo recommendation is that when set to **zfs** we set the install path to `zfs` and install our own `zfs.conf` which adds the search path:

```
> cat /etc/depmod.d/zfs.conf
search zfs
```

This way we do the right thing for distributions which support `extra`, and provide an override which can be used in the packaging of all distributions which don't.

Reply:
So Ubuntu already has:
```
$ cat /etc/depmod.d/ubuntu.conf
search updates ubuntu built-in
```
It's unclear to me from the docs what would happen if we add another `.conf` file with a `search` command.
For Ubuntu we could either add `zfs.conf` with an override statement for each zfs module, or modify `/etc/depmod.d/ubuntu.conf`.
The latter would be less error prone (since we may forget to add new modules) but might not be acceptable.

Reply:
If we want to ensure our modules are loaded, we could also define in our configuration `override` stanzas so that all modules produced by ZFS will _only_ get loaded from us.

Reply:
It does look like an `override` stanza would be needed.

@pzakha I'm OK with dropping this PR if you'd rather since the scope of what's needed has expanded a fair bit.  Though we will want something like this eventually so each downstream distribution doesn't need to carry a local patch to handle this.